### DISCUSSION TITLE
Updated Nue WL-SD001-9W

### DISCUSSION THREAD

**Discussion by Drulzy** [Koenkk/zigbee2mqtt#12257] [Devices]
Team, I recently got a new Nue WL-SD001-9W (9w 90mm RGB) in Australia. It has a different zigbeeModel of FEB56-ZCW2CLX1.0 compared with the already made LXN60-LS27-Z30

If I follow the support-new-devices guide to add it as a new profile it links to the current model page as the model ID is identical. 

Is it worth persisting with the new-devices guide, or can I try adding FEB56-ZCW2CLX1.0 to the existing Nue profile?

**Comment by Koenkk** [comment]
Try adding it to the existing definition.

**Comment by Drulzy** [comment]
Thanks. I'm going to stop pretending I know what I'm doing. I can see what needs to be changed in zigbee-herdsman-converters/devices/nue_3a.js, but I can't figure out how to deploy it to Zigbee2MQTT on Home Assistant, or to access the file in the Z2M instance. 

I'll keep bashing away at the long hand way, but I fear I'm out of my depth there, too...

**Comment by Drulzy** [comment]
Well, I was about to give up and ask for help. But I got it working. Long story short, all I did was create a new device as per  https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#instructions and copy-paste the original settings from https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/nue_3a.js

The new js looks like 
```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['FEB56-ZCW2CLX1.0'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'WL-SD001-9W-20220424', // Vendor model number, look on the device for a model number
    vendor: 'Nue / 3A', // Vendor of the device (only used for documentation and startup logging)
    description: '9W RGB LED downlight Aus 2022-04-24', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    extend: extend.light_onoff_brightness_colortemp_color({colorTempRange: [153, 370]}),
};

module.exports = definition;
```
Note that I added the date to the model ID as I wasn't sure if 2 having the same ID would be a problem. 

So I think herdsman should look like:
```
    {
        zigbeeModel: ['LXN60-LS27-Z30', 'FEB56-ZCW2CLX1.0'],
        model: 'WL-SD001-9W',
        vendor: 'Nue / 3A',
        description: '9W RGB LED downlight',
        extend: extend.light_onoff_brightness_colortemp_color({colorTempRange: [153, 370]}),
    },
```
but I don't know how to do the pull request. I'm not even beginner level on gits, I just don't understand them unless my programming environment automates it for me. So I guess I just don't understand them.

**Reply by Koenkk** [reply]
Added, thanks!

**Reply by Drulzy** [reply]
and detected correctly... thanks Koen.

**Comment by iotinkognito** [comment]
Did you have connectivity issues? I find it works when first powered and then disappears after that and there is no longer control etc.

**Reply by Drulzy** [reply]
yes, there is something going on... I bought 3 more and put them in a room. Intermittently one or more goes offline requiring a power cycle. idk if it's device or protocol related. 


`Error 2022-05-18 15:05:59Publish 'set' 'state' to 'DDDD Wardrobe Light' failed: 'Error: Command 0xae2f0211e00bfdb0/1 genOnOff.on({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received)'`

`Debug 2022-05-18 15:05:59Error: Command 0xae2f0211e00bfdb0/1 genOnOff.on({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (no response received) at DeconzAdapter.sendZclFrameToEndpoint (/app/node_modules/zigbee-herdsman/src/adapter/deconz/adapter/deconzAdapter.ts:649:23) at Endpoint.sendRequest (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:299:20) at Endpoint.command (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:772:28) at Object.convertSet (/app/node_modules/zigbee-herdsman-converters/converters/toZigbee.js:284:17) at Object.convertSet (/app/node_modules/zigbee-herdsman-converters/converters/toZigbee.js:866:40) at Publish.onMQTTMessage (/app/lib/extension/publish.ts:246:36)`

`Info 2022-05-18 15:05:59MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Publish 'set' 'state' to 'DDDD Wardrobe Light' failed: 'Error: Command 0xae2f0211e00bfdb0/1 genOnOff.on({}, {\"sendWhen\":\"immediate\",\"timeout\":10000,\"disableResponse\":false,\"disableRecovery\":false,\"disableDefaultResponse\":false,\"direction\":0,\"srcEndpoint\":null,\"reservedBits\":0,\"manufacturerCode\":null,\"transactionSequenceNumber\":null,\"writeUndiv\":false}) failed (no response received)'","meta":{"friendly_name":"DDDD Wardrobe Light"},"type":"zigbee_publish_error"}'`

**Reply by natdm** [reply]
Ever solve this? Same issue here. What firmware do you have? 

My specs:
```
Firmware build date
20210813
Firmware version
1.0.4
Manufacturer
[Nue / 3A](https://www.zigbee2mqtt.io/supported-devices/#v=Nue%20%2F%203A)
Model
[WL-SD001-9W](https://www.zigbee2mqtt.io/devices/WL-SD001-9W.html#nue%2520%252F%25203a-wl-sd001-9w)
```

**Comment by natdm** [comment]
I've got the same issue:
```
Error 2022-12-29 16:14:39Publish 'set' 'state' to 'Right Cabinet Lights' failed: 'Error: Command 0xed1e0310e00bfdb0/1 genOnOff.on({}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 27203 - 1 - 73 - 6 - 11 after 10000ms)'
```

The device works for a few minutes, then it stops and gets this error. Not sure what the deal is. I've got 3 devices and they all stop at the same time.

**Comment by Drulzy** [comment]
Hi mate.

I literally just got my lights back from Kevin at 3A yesterday with a new firmware. I have not tested them yet but I will fire them up tonight and post it here.

On 30 Dec 2022, at 11:40 am, Nathan Hyland ***@***.***> wrote:

﻿

Ever solve this? Same issue here. What firmware do you have?

My specs:

Firmware build date
20210813
Firmware version
1.0.4
Manufacturer
[Nue / 3A](https://www.zigbee2mqtt.io/supported-devices/#v=Nue%20%2F%203A)
Model
[WL-SD001-9W](https://www.zigbee2mqtt.io/devices/WL-SD001-9W.html#nue%2520%252F%25203a-wl-sd001-9w)

**Comment by Drulzy** [comment]
@natdm @iotinkognito 

After sending my lamps back to 3A they came back this week with 

Zigbee Model LXN60-LS27-Z30
Zigbee Manufacturer 3A Smart Home DE
Description 9W RGB LED downlight
Support status Supported

Firmware build date 20210813
Firmware version 1.1.7
Manufacturer [Nue / 3A](https://www.zigbee2mqtt.io/supported-devices/#v=Nue%20%2F%203A)

Note that the firmware version is higher, but the build date is the same. Importantly, the Zigbee Model is now the same as what Z2M had originally, not the FEB56 model ID . So hopefully this sorts them out. 

For what it's worth, I bought these on Amazon Australia, and communicated with 3A via the Amazon app. Kevin sent me his address in Victoria and had them back to me in less than 2 weeks over the Christmas break. Like 4 or 5 working days max.

**Reply by duk242** [reply]
Did this fix them? I'm having a similar issue where they fall off the network after ~3minutes

**Reply by Drulzy** [reply]
Indeed it did. I tried various scenarios including longer and longer inactivity. Each time the light responded within an appropriate delay compared to other devices on the mesh. I think I left them idle for > 3 days in the end, and they still responded.
